### PR TITLE
ovl_Obj_Taru Decompiled, object documented

### PR DESCRIPTION
**Pull request by sonicdcer**

### CODE REVIEW

**Review by AngheloAlf — APPROVED** [review]

File: src/overlays/actors/ovl_Obj_Taru/z_obj_taru.c [hunk 1]
```diff
@@ -28,55 +36,306 @@ const ActorInit Obj_Taru_InitVars = {
...
+    }
+}
+
+void func_80B9BCBC(ObjTaru* this, PlayState* play) {
+    s32 pad;
+    s32 params1F;
+    Actor* spawnedActor;
+    s16 rotY;
+
+    if (func_80B9B6E0(this, play) != 0) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_Obj_Taru/z_obj_taru.c:173]:
```suggestion
    if (func_80B9B6E0(this, play)) {
```

**Review by hensldm — APPROVED** [review]

File: src/overlays/actors/ovl_Obj_Taru/z_obj_taru.c [hunk 2]
```diff
@@ -28,55 +36,306 @@ const ActorInit Obj_Taru_InitVars = {
...
+    if (!OBJ_TARU_GET_80(thisx)) {
+        Collider_DestroyCylinder(play, &this->collider);
+    }
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Taru/func_80B9BCBC.s")
+s32 func_80B9BF7C(ObjTaru* this) {
+    s32 pad;
+    s32 phi_a3 = 0;
```
Comment (tom-overton) [inline src/overlays/actors/ovl_Obj_Taru/z_obj_taru.c:241]: This looks like a boolean:

```suggestion
    s32 phi_a3 = false;
```

File: src/overlays/actors/ovl_Obj_Taru/z_obj_taru.c [hunk 3]
```diff
@@ -28,55 +36,306 @@ const ActorInit Obj_Taru_InitVars = {
...
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Taru/func_80B9BD84.s")
+    if ((!OBJ_TARU_GET_80(&this->dyna.actor)) && (this->collider.base.acFlags & AC_HIT)) {
+        Actor* ac = this->collider.base.ac;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Taru/ObjTaru_Init.s")
+        this->collider.base.acFlags &= ~AC_HIT;
+        if (ac != NULL) {
+            if (this->collider.info.acHitInfo->toucher.dmgFlags & 0x80000000) {
+                phi_a3 = 0;
```
Comment (tom-overton) [inline src/overlays/actors/ovl_Obj_Taru/z_obj_taru.c:249]:
```suggestion
                phi_a3 = false;
```

File: src/overlays/actors/ovl_Obj_Taru/z_obj_taru.c [hunk 4]
```diff
@@ -28,55 +36,306 @@ const ActorInit Obj_Taru_InitVars = {
...
+                if (Math3D_Vec3fDistSq(&this->dyna.actor.world.pos, &ac->world.pos) < SQ(160.0f)) {
+                    phi_a3 = 1;
```
Comment (tom-overton) [inline src/overlays/actors/ovl_Obj_Taru/z_obj_taru.c:251]:
```suggestion
                    phi_a3 = true;
```

File: src/overlays/actors/ovl_Obj_Taru/z_obj_taru.c [hunk 5]
```diff
@@ -28,55 +36,306 @@ const ActorInit Obj_Taru_InitVars = {
...
+                }
+            } else if (this->collider.info.acHitInfo->toucher.dmgFlags & 8) {
+                if (Math3D_Vec3fDistSq(&this->dyna.actor.world.pos, &ac->world.pos) < SQ(100.0f)) {
+                    phi_a3 = 1;
```
Comment (tom-overton) [inline src/overlays/actors/ovl_Obj_Taru/z_obj_taru.c:255]:
```suggestion
                    phi_a3 = true;
```

File: src/overlays/actors/ovl_Obj_Taru/z_obj_taru.c [hunk 6]
```diff
@@ -28,55 +36,306 @@ const ActorInit Obj_Taru_InitVars = {
...
+                }
+            } else if (this->collider.info.acHitInfo->toucher.dmgFlags & 0x500) {
+                phi_a3 = 1;
```
Comment (tom-overton) [inline src/overlays/actors/ovl_Obj_Taru/z_obj_taru.c:258]:
```suggestion
                phi_a3 = true;
```

File: src/overlays/actors/ovl_Obj_Taru/z_obj_taru.c [hunk 7]
```diff
@@ -28,55 +36,306 @@ const ActorInit Obj_Taru_InitVars = {
...
+            }
+        }
+    } else if (this->dyna.actor.home.rot.z != 0) {
+        phi_a3 = 1;
```
Comment (tom-overton) [inline src/overlays/actors/ovl_Obj_Taru/z_obj_taru.c:262]:
```suggestion
        phi_a3 = true;
```

**Review by tom-overton — APPROVED** [review]
Might look like I left a lot of comments, but it's literally just all about one variable

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**